File: atcoder/abc317A_Potions.cpp
// Vicfred
// https://atcoder.jp/contests/abc317/tasks/abc317_a
// implementation
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int N, H, X;
  cin >> N >> H >> X;
  vector<int> P(N);
  for(int i = 0; i < N; ++i) {
    cin >> P[i];
  }
  for(int i = 0; i < N; ++i) {
    if(H + P[i] >= X) {
      cout << i + 1 << endl;
      return 0;
    }
  }
  return 0;
}

File: atcoder/abc317a_potions.hs
-- vicfred
-- https://atcoder.jp/contests/abc317/tasks/abc317_a
-- implementation
import Data.List (findIndex)
import Data.Maybe (fromMaybe)

main :: IO ()
main = do
  [n, h, x] <- map read . words <$> getLine :: IO [Int]
  p <- map read . words <$> getLine :: IO [Int]
  print . (+1) . fromMaybe (n - 1) . findIndex ((>=x) . (+h)) $ p
